Source Code/Visual C++ Project/Utilities.c

```diff
@@ -118,17 +119,21 @@ void PA_SetTextParameter(PA_PluginParameters params, short index, char* text, LO
 {
 	PA_Unistring		*UnistringParam;
 	PA_Unichar			*translatedText;
-	LONG_PTR			length;
+	LONG_PTR			length; 
+	size_t              convertedChars; // ZRW 11/17/16 WIN-39
+	wchar_t             junk;  // ZRW 12/1/16 WIN-39
 
 	if (len > 0){
-		length = mbstowcs(0, text, len) + 1;
-
-		translatedText = malloc(length * sizeof(PA_Unichar));
+		// ZRW 11/17/16 WIN-39 mbstowcs -> mbstowcs_s
+		mbstowcs_s(&convertedChars, &junk, 255, text, len);
+		length = convertedChars + 1;  
+	
+		translatedText = malloc(length * sizeof(PA_Unichar));  
```

Comment:
We don't need to multiply by sizeof here. Malloc(n) already allocates n bytes. We need length bytes, not length * sizeof(PA_Unichar) bytes. Not that this isn't the way it's always been, but since you're in here cleaning it up, it's not a bad idea to change this.